lib/controllers/register.js

```diff
@@ -125,159 +125,174 @@ module.exports = function (req, res) {
-      return res.sendFile(config.web.spaRoot);
-    }
+            helpers.expandAccount(req.app, account, function (err, expandedAccount) {
+              req.user = expandedAccount;
 
-    // If we get here, it means the user is doing a simple GET request, so we
-    // should just render the registration template.
-    if (req.method === 'GET') {
-      return helpers.render(req, res, view, {
-        form: helpers.sanitizeFormData(req.body, config),
-        formModel: formModel
-      });
-    }
+              if (config.web.register.autoLogin) {
+                return authenticator.authenticate({
+                  username: req.body.email || uuid(),
+                  password: req.body.password || uuid()
+                }, function (err, passwordGrantAuthenticationResult) {
+                  if (err) {
+                    return defaultJsonErrorResponder(err, res);
+                  }
 
-    // If we aren't getting a POST, we should bail quickly.
-    if (req.method !== 'POST') {
-      return res.status(415).end();
-    }
+                  helpers.createSession(passwordGrantAuthenticationResult, expandedAccount, req, res);
+
+                  if (postRegistrationHandler) {
+                    return postRegistrationHandler(expandedAccount, req, res, defaultJsonResponse.bind(null, req, res));
+                  }
+
+                  defaultJsonResponse(req, res);
+                });
+              }
 
-    // If we get here, it means the user is submitting a registration request, so
-    // we should attempt to validate the user's data and create their account.
-    async.waterfall([
-      // What we'll do here is simply set default values for `givenName` and
-      // `surname`, because these value are annoying to set if you don't
-      // care about them.  Eventually Stormpath is going to remove these
-      // required fields, but for now this is a decent workaround to ensure
-      // people don't have to deal with that stuff.
-      function (callback) {
-        applyDefaultAccountFields(config, req);
-        callback();
-      },
-      function (callback) {
-        helpers.validateAccount(req.body, req.app.get('stormpathConfig'), function (errors) {
-          if (errors) {
-            logger.info(errors);
-            return helpers.render(req, res, view, {
-              errors: errors,
-              form: helpers.sanitizeFormData(req.body, config),
-              formModel: formModel
+              if (postRegistrationHandler) {
+                return postRegistrationHandler(expandedAccount, req, res, defaultJsonResponse.bind(null, req, res));
+              }
+
+              defaultJsonResponse(req, res);
             });
+          });
+        });
+      });
+    },
+    // Handle incoming POST requests from an browser-like clients (something like
+    // chrome / firefox / etc.).
+    'text/html': function () {
+      switch (req.method) {
+        // We should render the registration template.
```

Comment:
With #331 , the `handleAcceptRequest` is now serving the SPA root, so this can go away.